File: api/models/network.py
# api/models/network.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class NetworkData(BaseModel):
    id: int
    ip_address: Optional[str] = None
    mac_address: Optional[str] = None
    hostname: Optional[str] = None
    manufacturer: Optional[str] = None
    device_type: Optional[str] = None
    whitelist: Optional[bool] = False
    status: Optional[str] = "Active"
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    
    class Config:
        orm_mode = True

File: final/src/app/lib/network.ts
export interface Networks{
    id: number;
    ip_address: string;
    mac_address: string;
    hostname: string;
    manufacturer: string;
    whitelist: number;
    status: string;
    created_at: string;
    updated_at: string;
}

export interface NetworkWithBandwidth{
    id: number;
    ip_address: string;
    mac_address: string;
    hostname: string;
    manufacturer: string;
    whitelist: number;
    device_type: string;
    status: string;
    created_at: string;
    updated_at: string;
    download: number;
    upload: number;
    bandwidth_timestamp: string;
}


export interface UnblockProp{
    id: number;
    status: string;
}

